plugin/src/main/java/org/opensearch/ml/rest/RestMLDeleteMemoryContainerAction.java

```diff
@@ -62,11 +69,48 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
         }
 
         String memoryContainerId = request.param(PARAMETER_MEMORY_CONTAINER_ID);
-        boolean deleteAllMemories = request.paramAsBoolean(PARAMETER_DELETE_ALL_MEMORIES, false);
         String tenantId = getTenantID(mlFeatureEnabledSetting.isMultiTenancyEnabled(), request);
+
+        // Parse URL parameters
+        boolean deleteAllMemories = request.paramAsBoolean(PARAMETER_DELETE_ALL_MEMORIES, false);
+        Set<String> deleteMemories = null;
```

Comment:
Use MemoryType enum ? 
```suggestion
        Set<MemoryType> deleteMemories = null;
```